src/core/trulens/core/utils/python.py

```diff
@@ -437,10 +445,10 @@ def external_caller_frame(offset=0) -> FrameType:
     Raises:
         RuntimeError: If no such frame is found.
     """
-
-    for finfo in inspect.stack()[offset + 1 :]:
-        if not finfo.frame.f_globals["__name__"].startswith("trulens"):
-            return finfo.frame
+    gen = list(stack_generator(offset + 2))
```

Comment:
casting to list shouldn't be necessary here right? and it's more memory efficient if you just iterate over the generator without forcing to evaluate the entire stack all at once

Reply:
I was originally thinking the same. Then realized that because the generator is called lazily, its call stack would change and reflect the generator callee's call stack, not the call stack of `external_caller_frame`

Reply:
ah got it sg

Reply:
actually we may be able to get around this by passing in the current frame so that inspect.currentframe() is guarenteed to be invoked at the right time